packages/ember-metal/lib/computed.js

```diff
@@ -30,80 +30,104 @@ function UNDEFINED() { }
 //
 
 /**
-  A computed property transforms an object's function into a property.
-
+  A computed property transforms a hash with object's accessor function(s) into a property.
+  
   By default the function backing the computed property will only be called
   once and the result will be cached. You can specify various properties
   that your computed property depends on. This will force the cached
   result to be recomputed if the dependencies are modified.
-
-  In the following example we declare a computed property (by calling
-  `.property()` on the fullName function) and setup the property
-  dependencies (depending on firstName and lastName). The fullName function
+  
+  In the following example we declare a computed property - `fullName` - by calling
+  `.Ember.computed()` with property dependencies (`firstName` and `lastName`) as leading arguments and getter accessor function. The `fullName` getter function
   will be called once (regardless of how many times it is accessed) as long
-  as its dependencies have not changed. Once firstName or lastName are updated
-  any future calls (or anything bound) to fullName will incorporate the new
+  as its dependencies have not changed. Once `firstName` or `lastName` are updated
+  any future calls (or anything bound) to `fullName` will incorporate the new
   values.
-
+  
   ```javascript
-  var Person = Ember.Object.extend({
+  let Person = Ember.Object.extend({
     // these will be supplied by `create`
     firstName: null,
     lastName: null,
-
-    fullName: function() {
-      var firstName = this.get('firstName');
-      var lastName = this.get('lastName');
-
-     return firstName + ' ' + lastName;
-    }.property('firstName', 'lastName')
+  
+    fullName: Ember.computed('firstName', 'lastName', function () {
+      let firstName = this.get('firstName'),
+          lastName  = this.get('lastName');
```

Comment:
We should follow our styleguide here, we prefer a declarations with assignment to be stand alone statements.